### PR TITLE
Decouple Theme

### PR DESCRIPTION
**Pull request by JugglerX**

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (JugglerX) [inline .eslintrc.json:1]: @smnh any reason to change the linting? Are these rules exactly the same? If not, you need to ask the team and get a majority before changing linting rules.

File: src/css/main.css [hunk 2]
```diff
@@ -1,5 +1,353 @@
-@import '@stackbit/components/styles/default.css';
-@import "tailwindcss/base";
-@import "tailwindcss/components";
-@import "tailwindcss/utilities";
-@import 'theme.css';
\ No newline at end of file
+@import url('https://fonts.googleapis.com/css2?family=Noto+Serif:ital,wght@0,400;0,700;1,400;1,700&family=Poppins:ital,wght@0,400;0,700;1,400;1,700&display=swap');
+
+@tailwind base;
```
Comment (JugglerX) [inline src/css/main.css:3]: @smnh why did you change it from @import?

File: tailwind.config.js [hunk 3]
```diff
@@ -2,7 +2,16 @@ const plugin = require('tailwindcss/plugin');
 const themeStyle = require('./content/data/style.json');
 
 module.exports = {
-    presets: [require('@stackbit/components/styles/tailwind.default.config.js')],
+    mode: 'jit',
+    purge: {
+        enabled: false,
```
Comment (JugglerX) [inline tailwind.config.js:7]: should be true

File: src/pages/_app.js [hunk 4]
```diff
@@ -1,14 +1,5 @@
-import '../components/register-components';
 import '../css/main.css';
-import Head from 'next/head';
 
 export default function MyApp({ Component, pageProps }) {
-    return (
```
Comment (JugglerX) [inline src/pages/_app.js:6]: @smnh We ned this and I cant find where you redefined the `<Head>`.

File: src/css/main.css [hunk 5]
```diff
@@ -1,5 +1,353 @@
...
+        }
+    }
+
+    /* Buttons */
+    .sb-component-button {
+        @apply border-2 inline-flex items-center justify-center no-underline text-center transition duration-200 ease-in hover:opacity-80;
+    }
+
+    /* Links */
+    .sb-component-link {
```
Comment (JugglerX) [inline src/css/main.css:136]: @smnh @TomasBankauskas `.sb-component-link` is partially defined here in the CSS but also in the `tailwind.config.js` inside `addComponents` - Does it make sense to have it in both places?

Same thing applies to a few other classes.

File: next.config.js [hunk 6]
```diff
@@ -1,4 +1,3 @@
-const path = require('path');
 const sourcebit = require('sourcebit');
 const sourcebitConfig = require('./sourcebit.js');
 const withBundleAnalyzer = require('@next/bundle-analyzer')({
```
Comment (JugglerX) [inline next.config.js:3]: @smnh we should remove BundleAnalyzer

File: src/css/main.css [hunk 7]
```diff
@@ -1,5 +1,353 @@
...
+    .sb-select {
+        @apply appearance-none bg-no-repeat cursor-pointer pr-7;
+        background-image: linear-gradient(45deg, transparent 50%, currentColor 0), linear-gradient(135deg, currentColor 50%, transparent 0);
+        background-position: calc(100% - 16px) calc(1px + 50%), calc(100% - 12px) calc(1px + 50%);
+        background-size: 4px 4px, 4px 4px;
+        background-repeat: no-repeat;
+    }
+
+    /* Color palettes */
+    .colors-a {
```
Comment (JugglerX) [inline src/css/main.css:173]: @smnh @TomasBankauskas do we still need all the colors for this theme? If not delete them from here and also in the tailwind purge array.

File: .eslintrc.json (hunk 1 above)
Comment (smnh) [reply .eslintrc.json:1]: Not really, it is still TODO, and I also want to use next's own linting rules so it will be very natural to next.js users. It is like respecting the Next.js and its conventions. But we don't have to. Let's think about that. I still need to add some additional rules there from old components package.

File: src/css/main.css (hunk 2 above)
Comment (smnh) [reply src/css/main.css:3]: this is a standard way to use tailwind no?

Comment (smnh) [reply src/css/main.css:3]: It what we had in the default.css file in components.
the theme then imported it.
but if everything now is in single file, why not to use the standard

File: tailwind.config.js (hunk 3 above)
Comment (smnh) [reply tailwind.config.js:7]: yes, I copy pasted it, it was false for some reason

File: src/pages/_app.js (hunk 4 above)
Comment (smnh) [reply src/pages/_app.js:6]: favicons is in base layouts like everything else, the _app.js should be used for other things, not for adding favicons.

File: next.config.js (hunk 6 above)
Comment (smnh) [reply next.config.js:3]: actually I thought leaving it there and explaining developers in docs to run it to validate their bundle size.
not critical tho.
also we need to add `next link` to scripts because it is Next.js standard and their new convention that checks for important next.js things.

File: src/css/main.css (hunk 7 above)
Comment (smnh) [reply src/css/main.css:173]: Yes, the css and all these stuffs are up to you, I just copy pasted everything

File: src/css/main.css (hunk 5 above)
Comment (smnh) [reply src/css/main.css:136]: Agree.
@TomasBankauskas @JugglerX I suggest simplifying it as much as possible. If we think defining component classes is simpler and intuitive in css files then lets have them only there.
On the other hand, if there are things that are configurable via Stackbit UI like h1, h2, etc., then these should be in tailwind.config.js because it is easier to configure them when they are in js file.
